bindings/wasm/src/account/traits.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::fmt::Debug;
+use core::fmt::Formatter;
+
+use js_sys::Promise;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::JsFuture;
+
+use identity::account::ChainState;
+use identity::account::DIDLease;
+use identity::account::EncryptionKey;
+use identity::account::Generation;
+use identity::account::IdentityState;
+use identity::account::KeyLocation;
+use identity::account::Result as AccountResult;
+use identity::account::Signature;
+use identity::account::Storage;
+use identity::core::encode_b58;
+use identity::crypto::PrivateKey;
+use identity::crypto::PublicKey;
+use identity::iota::IotaDID;
+
+use crate::account::WasmChainState;
+use crate::account::WasmEncryptionKey;
+use crate::account::WasmGeneration;
+use crate::account::WasmIdentityState;
+use crate::account::WasmKeyLocation;
+use crate::did::WasmDID;
+use crate::error::JsValueResult;
+
+#[wasm_bindgen]
+extern "C" {
+  pub type WasmStorage;
+  #[wasm_bindgen(typescript_type = "Promise<()>")]
+  pub type PromiseUnit;
+  #[wasm_bindgen(typescript_type = "Promise<DIDLease>")]
+  pub type PromiseDIDLease;
+  #[wasm_bindgen(typescript_type = "Promise<PublicKey>")]
+  pub type PromisePublicKey;
+  #[wasm_bindgen(typescript_type = "Promise<Signature>")]
+  pub type PromiseSignature;
+  #[wasm_bindgen(typescript_type = "Promise<bool>")]
+  pub type PromiseBool;
+  #[wasm_bindgen(typescript_type = "Promise<Option<Generation>>")]
+  pub type PromiseOptionGeneration;
+  #[wasm_bindgen(typescript_type = "Promise<Option<ChainState>>")]
+  pub type PromiseOptionChainState;
+  #[wasm_bindgen(typescript_type = "Promise<Option<IdentityState>>")]
+  pub type PromiseOptionIdentityState;
```

Comment:
Nit: maybe move the `Promise` definitions to a separate `extern "C"` block to make it easier to see the `WasmStorage` definitions.